### PR TITLE
Improve buf-ci action and add README and LICENSE

### PR DESCRIPTION
**Pull request by unmultimedio**
We might have other `LICENSE/README/buf.yaml` files for static config for individual managed modules, we don't want to trigger the `buf-ci.yaml` action when changing those, only when changing something within the `proto/*` dir.

Add symlinks/content for `LICENSE/README` in the base BSR module.

### CODE REVIEW

File: .github/workflows/buf-ci.yaml [hunk 1]
```diff
@@ -2,15 +2,17 @@ name: Buf CI
...
     paths:
-      - '**.proto'
-      - '**/buf.yaml'
-      - '**/buf.lock'
-      - '**/buf.md'
-      - '**/README.md'
-      - '**/LICENSE'
-      - .github/workflows/buf-ci.yaml
+      - 'proto/**.proto'
+      - 'proto/**/buf.yaml'
```
Comment (pkwarren) [inline .github/workflows/buf-ci.yaml:6]:
```suggestion
      - 'buf.yaml'
```

File: .github/workflows/buf-ci.yaml [hunk 2]
```diff
@@ -2,15 +2,17 @@ name: Buf CI
...
+      - 'proto/**/buf.lock'
```
Comment (pkwarren) [inline .github/workflows/buf-ci.yaml:7]:
```suggestion
      - 'buf.lock'
```

File: .github/workflows/buf-ci.yaml [hunk 3]
```diff
@@ -2,15 +2,17 @@ name: Buf CI
...
+      - 'proto/**/buf.md'
```
Comment (pkwarren) [inline .github/workflows/buf-ci.yaml:8]:
```suggestion
      - 'buf.md'
```

File: .github/workflows/buf-ci.yaml [hunk 4]
```diff
@@ -2,15 +2,17 @@ name: Buf CI
...
+      - 'proto/**/README.md'
```
Comment (pkwarren) [inline .github/workflows/buf-ci.yaml:9]:
```suggestion
      - 'README.md'
```

File: .github/workflows/buf-ci.yaml [hunk 5]
```diff
@@ -2,15 +2,17 @@ name: Buf CI
...
+      - 'proto/**/LICENSE'
```
Comment (pkwarren) [inline .github/workflows/buf-ci.yaml:10]:
```suggestion
      - 'LICENSE'
```

File: proto/LICENSE [hunk 6]
```diff
@@ -0,0 +1 @@
+../LICENSE
```
Comment (pkwarren) [inline proto/LICENSE:1]: We don't need these do we?

File: proto/README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+../README.md
```
[the remark this answers is not in the document]
Comment (unmultimedio) [reply proto/README.md:1]: Nice! I can revert these two when that lands 😄

File: .github/workflows/buf-ci.yaml (hunk 2 above)
Comment (emcfarlane) [inline .github/workflows/buf-ci.yaml:7]:
```suggestion
      - 'buf.yaml'
      - 'buf.lock'
```
Safe to presume this action only runs for the input defined at the root.

File: .github/workflows/buf-ci.yaml (hunk 3 above)
Comment (unmultimedio) [reply .github/workflows/buf-ci.yaml:8]: We don't have that file, I'd say let's just remove it?

File: .github/workflows/buf-ci.yaml (hunk 4 above)
Comment (emcfarlane) [reply .github/workflows/buf-ci.yaml:8]: Nit: should be `proto/README.md`, don't think any other would affect it.

**Review by emcfarlane — APPROVED** [review]

File: .github/workflows/fetch.yaml [hunk 8]
```diff
@@ -44,6 +44,7 @@ jobs:
         with:
           add-paths: ./modules/sync/**
           commit-message: "Detected new managed modules references"
+          # This branch is ignored for the 'buc-fi.yaml' action. Keep this branch name synced there.
```
Comment (pkwarren) [inline .github/workflows/fetch.yaml:47]:
```suggestion
          # This branch is ignored for the 'buf-ci.yaml' action. Keep this branch name synced there.
```

**Comment by unmultimedio** [comment]
https://buf.build/bufbuild/managed-modules/docs/jfigueroa/buf-ci-improvements

**Review by pkwarren — APPROVED** [review]

**Pull request merged**